### PR TITLE
Icons enhancement

### PR DESCRIPTION
**Pull request by maweki**
Is there anything more to do before merging?

I suggest we squash this whole thing.

### CODE REVIEW

**Comment by RaphaelRochet** [comment]
We should maybe beta test this for a week a two ?

**Comment by maweki** [comment]
Sure. If you see any issues or have nicer implementation ideas...

**Comment by maweki** [comment]
I just tried changing my twitch profile picture and it worked. As soon as I got online, the new picture with the new id was downloaded and the new one was symlinked as the correct one.
bildschirmfoto von 2016-06-06 14-39-16

**Comment by RaphaelRochet** [comment]
Nice !

Le lun. 6 juin 2016 à 14:39, Mario Wenzel <notifications@github.com> a
écrit :

File: extension.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ const Extension = imports.misc.extensionUtils.getCurrentExtension();
 const MenuItems = Extension.imports.menu_items;
 const Promise = Extension.imports.promise.Promise;
```
Comment (RaphaelRochet) [inline extension.js]: Your forgot this line I guess !

**Comment by RaphaelRochet** [comment]
It works flawlessly. Brilliant work ;) Merge when you want.

File: extension.js (hunk 1 above)
Comment (maweki) [inline extension.js]: Well, we're still doing `new Promise.all(requests).then(` a few lines later, which remained unchanged.

Comment (RaphaelRochet) [inline extension.js]: We shall move this to API then. But could be done later.

Comment (maweki) [inline extension.js]: Why? And how? This is not API behaviour but extension logic. Doing something once these specific api requests are done.

**Pull request merged**